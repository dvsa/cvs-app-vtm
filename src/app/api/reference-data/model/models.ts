export * from './reference-data.model';
