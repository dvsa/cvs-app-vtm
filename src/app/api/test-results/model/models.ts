export * from './carCancelled';
export * from './carCancelledAdditionalInformation';
export * from './carCancelledAdditionalInformationLocation';
export * from './carCancelledCustomDefects';
export * from './carCancelledDefects';
export * from './carCancelledModType';
export * from './carCancelledTestTypes';
export * from './carCancelledVehicleClass';
export * from './carSubmitted';
export * from './carSubmittedAdditionalInformation';
export * from './carSubmittedDefects';
export * from './carSubmittedTestTypes';
export * from './completeTestResults';
export * from './completeTestResultsVehicleClass';
export * from './customDefect';
export * from './customDefects';
export * from './defect';
export * from './defectAdditionalInformation';
export * from './defectAdditionalInformationLocation';
export * from './defects';
export * from './hgvCancelled';
export * from './hgvSubmitted';
export * from './lgvCancelled';
export * from './lgvSubmitted';
export * from './motorcycleCancelled';
export * from './motorcycleSubmitted';
export * from './motorcycleSubmittedVehicleClass';
export * from './msUserDetails';
export * from './psvCancelled';
export * from './psvCancelledModType';
export * from './psvCancelledTestTypes';
export * from './psvSubmitted';
export * from './psvSubmittedAdditionalInformation';
export * from './psvSubmittedAdditionalInformationLocation';
export * from './psvSubmittedDefects';
export * from './psvSubmittedTestTypes';
export * from './testHistory';
export * from './testResultPutBody';
export * from './testResults';
export * from './testTypeRecords';
export * from './testTypeResults';
export * from './testTypeResultsModType';
export * from './testTypes';
export * from './trlCancelled';
export * from './trlCancelledTestTypes';
export * from './trlSubmitted';
export * from './trlSubmittedTestTypes';
