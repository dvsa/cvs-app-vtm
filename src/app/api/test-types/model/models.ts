export * from './testType';
export * from './testTypeCategory';
export * from './testTypeInfo';
export * from './testTypesTaxonomy';
