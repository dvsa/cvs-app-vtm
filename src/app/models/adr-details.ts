import { Tank } from './Tank';

interface VehicleDetails {
  type: string;
  approvalDate: string;
}

interface AdditionalNotes {
  number: string[];
  guidanceNotes: string[];
}

export interface ApplicantDetails {
  name: string;
  street: string;
  town: string;
  city: string;
  postcode: string;
}

export interface AdrDetails {
  vehicleDetails: VehicleDetails;
  listStatementApplicable: boolean;
  batteryListNumber: string;
  declarationsSeen: boolean;
  brakeDeclarationsSeen: boolean;
  brakeDeclarationIssuer: string;
  brakeEndurance: boolean;
  weight: string;
  compatibilityGroupJ?: boolean;
  documents: string[];
  permittedDangerousGoods: string[];
  additionalExaminerNotes: string;
  applicantDetails: ApplicantDetails;
  memosApply: string[];
  additionalNotes: AdditionalNotes;
  adrTypeApprovalNo: string;
  tank: Tank;
}
