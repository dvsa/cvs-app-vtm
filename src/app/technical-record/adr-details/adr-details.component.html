<!--  <dl class="technical-record-sub-details govuk-summary-list adr-details">-->
<!--    <div class="govuk-summary-list__row">-->
<!--      <dt class="govuk-summary-list__key">-->
<!--        Able to carry dangerous goods-->
<!--        <span *ngIf="showCheck">-->
<!--          <div class="govuk-radios govuk-radios&#45;&#45;inline" id="radioADR">-->
<!--            <div class="govuk-radios__item">-->
<!--              <input class="govuk-radios__input" id="hasADR" name="hasADR" type="radio" [checked]="!isAdrNull"-->
<!--                     value="true" (change)="switchAdrDisplay($event)">-->
<!--              <label class="govuk-label govuk-radios__label" for="hasADR">-->
<!--                Yes-->
<!--              </label>-->
<!--            </div>-->
<!--            <div class="govuk-radios__item">-->
<!--              <input class="govuk-radios__input" id="hasNotADR" name="hasADR" type="radio"-->
<!--                     [checked]="isAdrNull" value="false" (change)="switchAdrDisplay($event)">-->
<!--              <label class="govuk-label govuk-radios__label" for="hasNotADR">-->
<!--                No-->
<!--              </label>-->
<!--            </div>-->
<!--          </div>-->
<!--        </span>-->
<!--      </dt>-->

<!--      <dd id="test-ableToCarry" class="govuk-summary-list__value">-->
<!--        <div *ngIf="!showCheck">-->
<!--          <span> {{ activeRecord.adrDetails == null ? "No" : "Yes"  }}</span>-->
<!--        </div>-->
<!--      </dd>-->
<!--      <dd class="govuk-summary-list__value"></dd>-->
<!--    </div>-->

<!--    <vtm-adr-details-view>-->
<!--        [activeRecord]="activeRecord"-->
<!--        [techRecordsJson]="techRecordsJson"-->
<!--        [hideForm]="hideForm">-->
<!--    </vtm-adr-details-view>-->

<!--&lt;!&ndash;    <ng-template #adrForm>&ndash;&gt;-->
<!--&lt;!&ndash;        <vtm-adr-details-form *ngIf="!hideForm"&ndash;&gt;-->
<!--&lt;!&ndash;          [vehicletypes$]="vehicletypes$"&ndash;&gt;-->
<!--&lt;!&ndash;          [permittedDangerousGoodsFe$]="permittedDangerousGoodsFe$"&ndash;&gt;-->
<!--&lt;!&ndash;          [guidanceNotesFe$]="guidanceNotesFe$"&ndash;&gt;-->
<!--&lt;!&ndash;          [initialAdrDetails]="activeRecord?.adrDetails">&ndash;&gt;-->
<!--&lt;!&ndash;        </vtm-adr-details-form>&ndash;&gt;-->
<!--&lt;!&ndash;    </ng-template>&ndash;&gt;-->

<!--  </dl>-->
