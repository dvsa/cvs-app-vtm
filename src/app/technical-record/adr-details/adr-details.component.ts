import {Component, OnInit, ViewEncapsulation, ChangeDetectionStrategy, Input} from '@angular/core';
import {Observable} from 'rxjs';
import {select, Store} from '@ngrx/store';
import {IAppState} from '@app/technical-record/adr-details/adr-details-form/store/adrDetailsForm.state';

@Component({
  selector: 'vtm-adr-details',
  templateUrl: './adr-details.component.html',
  styleUrls: ['../../app.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AdrDetailsComponent implements OnInit {

  @Input() activeRecord: any;
  @Input() showCheck: boolean;
  @Input() isAdrNull: boolean;
  @Input() adrData: boolean;
  @Input() hideForm: boolean;
  @Input() techRecordsJson: any;

  vehicletypes$: Observable<string[]>;
  permittedDangerousGoodsFe$: Observable<string[]>;
  guidanceNotesFe$: Observable<string[]>;

  constructor(private _store: Store<IAppState>) { }

  ngOnInit() {
    this.vehicletypes$ = this._store.pipe(select(s => s.vehicleTechRecordModel.vehicleTechRecordModel[0].metadata.adrDetails.vehicleDetails.typeFe));
    this.permittedDangerousGoodsFe$ = this._store.pipe(select(s => s.vehicleTechRecordModel.vehicleTechRecordModel[0].metadata.adrDetails.permittedDangerousGoodsFe));
    this.guidanceNotesFe$ = this._store.pipe(select( s => s.vehicleTechRecordModel.vehicleTechRecordModel[0].metadata.adrDetails.additionalNotes.guidanceNotesFe));
  }

  switchAdrDisplay($event): void {
    this.adrData = !($event.currentTarget.value === 'true');
    this.hideForm = $event.currentTarget.value === 'false';
  }

}
