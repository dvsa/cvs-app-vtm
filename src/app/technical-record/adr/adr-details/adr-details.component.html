<ng-container *ngIf="!edit">
  <h2 class="govuk-heading-m">ADR details</h2>

  <dl class="govuk-summary-list">
    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Vehicle type</dt>
      <dd id="test-ADR-type" class="govuk-summary-list__value">
        {{ adrDetails.vehicleDetails?.type | DefaultNullOrEmpty | titlecase }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Date processed</dt>
      <dd id="test-ADR-approvalDate" class="govuk-summary-list__value">
        {{ adrDetails.vehicleDetails?.approvalDate | DefaultNullOrEmpty }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Permitted dangerous goods</dt>
      <dd id="test-ADR-permittedDangerousGoods" class="govuk-summary-list__value">
        <div *ngIf="adrDetails.permittedDangerousGoods; else hyphen">
          <div *ngFor="let perGood of adrDetails.permittedDangerousGoods; let i = index">
            <p>
              <span id="test-ADR-permittedDangerousGoods-{{ i }}">{{ perGood }}</span>
            </p>
          </div>
        </div>
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Compatibility group J</dt>
      <dd id="test-ADR-compatibilityGroupJ" class="govuk-summary-list__value">
        {{ adrDetails?.compatibilityGroupJ | DefaultNullOrEmpty }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Guidance notes</dt>
      <dd id="test-ADR-additionalNotes-number" class="govuk-summary-list__value">
        <div
          *ngIf="
            this.adrDetails.additionalNotes && adrDetails.additionalNotes.number[0];
            else hyphen
          "
        >
          <div *ngFor="let number of adrDetails.additionalNotes.number; let i = index">
            <p>
              <span id="test-ADR-additionalNotes-number-{{ i }}">{{ number }}</span>
            </p>
          </div>
        </div>
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">ADR type approval number</dt>
      <dd id="test-ADR-adrTypeApprovalNo" class="govuk-summary-list__value">
        {{ adrDetails.adrTypeApprovalNo | DefaultNullOrEmpty }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>
  </dl>
</ng-container>

<ng-template #hyphen><span>-</span></ng-template>

<!-- Edit Applicant Details -->
<ng-container *ngIf="edit">
  <vtm-adr-details-edit [adrDetails]="adrDetails" [metaData]="metaData"></vtm-adr-details-edit>
</ng-container>
