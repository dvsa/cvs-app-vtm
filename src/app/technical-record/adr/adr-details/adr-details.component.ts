import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';

import { AdrDetails } from '@app/models/adr-details';
import { MetaData } from '@app/models/meta-data';

@Component({
  selector: 'vtm-adr-new-details',
  templateUrl: './adr-details.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AdrDetailsComponent implements OnInit {
  @Input() edit: boolean;
  @Input() adrDetails: AdrDetails;
  @Input() metaData: MetaData;

  constructor() {}

  ngOnInit() {}
}
