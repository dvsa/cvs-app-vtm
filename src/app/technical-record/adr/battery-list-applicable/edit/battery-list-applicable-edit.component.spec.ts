import { By } from '@angular/platform-browser';
import { CUSTOM_ELEMENTS_SCHEMA, DebugElement } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule, FormsModule, FormGroupDirective } from '@angular/forms';
import { of, EMPTY } from 'rxjs';

import { SharedModule } from '@app/shared';
import { TESTING_UTILS } from '@app/utils/testing.utils';
import { BatteryListApplicableEditComponent } from './battery-list-applicable-edit.component';
import { AdrDetails } from '@app/models/adr-details';
import { ValidationMapper, STATUS } from '../../adr-validation.mapper';

describe('BatteryListApplicableEditComponent', () => {
  let fixture: ComponentFixture<BatteryListApplicableEditComponent>;
  let component: BatteryListApplicableEditComponent;
  let validationMapper: ValidationMapper;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [FormsModule, ReactiveFormsModule, SharedModule],
      declarations: [BatteryListApplicableEditComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
      providers: [
        FormGroupDirective,
        {
          provide: FormGroupDirective,
          useValue: TESTING_UTILS.mockFormGroupDirective()
        }
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    validationMapper = TestBed.get(ValidationMapper);
    fixture = TestBed.createComponent(BatteryListApplicableEditComponent);
    component = fixture.componentInstance;
    component.adrDetails = TESTING_UTILS.mockAdrDetails({
      listStatementApplicable: true,
      batteryListNumber: 'bat123'
    } as AdrDetails);
  });

  it('should create with initialized form controls', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should setup a default battery list reference number', () => {
    fixture.detectChanges();

    const expectedValue = component.adrForm.get('batteryListNumber').value;
    expect(expectedValue).toEqual('bat123');
  });

  describe('handleFormChanges', () => {
    let ele: DebugElement;

    beforeEach(() => {
      ele = fixture.debugElement;
    });

    it('should update the battery list reference number field', () => {
      fixture.detectChanges();

      const refInput: HTMLInputElement = ele.query(By.css('[id=batteryListNumber]'))
        .nativeElement;
      refInput.value = 'bat124';
      refInput.dispatchEvent(new Event('input'));

      const expectedValue = component.adrForm.get('batteryListNumber').value;
      expect(expectedValue).toEqual('bat124');
    });

    it('should reset entered ref number when "no" option is selected', () => {
      fixture.detectChanges();

      const selectedOption: HTMLElement = ele.query(By.css('[id="listStatementApplicable-No"]'))
        .nativeElement;

      selectedOption.click();

      expect(component.adrForm.get('batteryListNumber').value).toBeNull();
    });

    it('should hide section when vehicle type dispatches HIDDEN', () => {
      spyOn(validationMapper, 'getVehicleTypeState').and.callFake(() =>
        of({ batteryListApplicableEdit: STATUS.HIDDEN })
      );

      fixture.detectChanges();

      let batteryState: STATUS;
      component.batterListValidationState$.subscribe(
        ({ batteryListApplicableEdit }) => (batteryState = batteryListApplicableEdit)
      );

      expect(batteryState).toEqual(STATUS.HIDDEN);
      expect(fixture).toMatchSnapshot();
      expect(component.adrForm.get('listStatementApplicable').value).toBeNull();
      expect(component.adrForm.get('batteryListNumber').value).toBeNull();
    });

    it('should show section when selected vehicle type dispatches no status', () => {
      spyOn(validationMapper, 'getVehicleTypeState').and.callFake(() => of(EMPTY));

      fixture.detectChanges();

      let batteryState: STATUS;
      component.batterListValidationState$.subscribe(
        ({ batteryListApplicableEdit }) => (batteryState = batteryListApplicableEdit)
      );
      expect(component.showBatteryListSection).toBeTruthy();
    });
  });
});
