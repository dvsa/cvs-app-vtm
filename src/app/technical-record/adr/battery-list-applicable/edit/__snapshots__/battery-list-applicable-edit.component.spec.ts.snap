// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BatteryListApplicableEditComponent handleFormChanges should hide section when vehicle type dispatches HIDDEN 1`] = `
<vtm-battery-list-applicable-edit
  adrDetails={[Function Object]}
  adrForm={[Function FormGroup]}
  batterListValidationState$={[Function Observable]}
  detectChange={[Function ViewRef_]}
  fb={[Function FormBuilder]}
  onDestroy$={[Function Subject]}
  options={[Function Object]}
  parent={[Function FormGroupDirective]}
  removeAdrControl={[Function EventEmitter]}
  showBatteryListSection="false"
  validationMapper={[Function ValidationMapper]}
>
  
</vtm-battery-list-applicable-edit>
`;

exports[`BatteryListApplicableEditComponent should create with initialized form controls 1`] = `
<vtm-battery-list-applicable-edit
  adrDetails={[Function Object]}
  adrForm={[Function FormGroup]}
  batterListValidationState$={[Function Observable]}
  detectChange={[Function ViewRef_]}
  fb={[Function FormBuilder]}
  onDestroy$={[Function Subject]}
  options={[Function Object]}
  parent={[Function FormGroupDirective]}
  removeAdrControl={[Function EventEmitter]}
  showBatteryListSection={[Function Boolean]}
  validationMapper={[Function ValidationMapper]}
>
  <div
    class="govuk-grid-row"
  >
    <div
      class="govuk-grid-column-full"
    >
      <div
        class="govuk-form-group"
      >
        <fieldset
          class="govuk-fieldset"
        >
          <div
            class="ng-untouched ng-pristine ng-valid"
            ng-reflect-form="[object Object]"
          >
            <h2
              class="govuk-heading-m"
            >
              Battery list applicable
            </h2>
            <div
              class="govuk-radios govuk-radios--inline"
            >
              
              <div
                class="govuk-radios__item"
              >
                <input
                  class="govuk-radios__input ng-untouched ng-pristine ng-valid"
                  formcontrolname="listStatementApplicable"
                  id="listStatementApplicable-Yes"
                  name="listStatementApplicable"
                  ng-reflect-form-control-name="listStatementApplicable"
                  ng-reflect-name="listStatementApplicable"
                  ng-reflect-value="true"
                  type="radio"
                />
                <label
                  class="govuk-label govuk-radios__label"
                  for="listStatementApplicable-Yes"
                >
                   Yes 
                </label>
              </div>
              <div
                class="govuk-radios__item"
              >
                <input
                  class="govuk-radios__input ng-untouched ng-pristine ng-valid"
                  formcontrolname="listStatementApplicable"
                  id="listStatementApplicable-No"
                  name="listStatementApplicable"
                  ng-reflect-form-control-name="listStatementApplicable"
                  ng-reflect-name="listStatementApplicable"
                  ng-reflect-value="false"
                  type="radio"
                />
                <label
                  class="govuk-label govuk-radios__label"
                  for="listStatementApplicable-No"
                >
                   No 
                </label>
              </div>
              <div
                class="govuk-inset-text govuk-!-margin-top-0"
                id="conditional-battery-list-applicable"
              >
                <div
                  class="govuk-form-group govuk-!-margin-left-3"
                >
                  <label
                    class="govuk-label govuk-!-font-weight-bold"
                    for="batteryListNumber"
                  >
                     Reference number 
                  </label>
                  <input
                    class="govuk-input govuk-input--width-10 ng-untouched ng-pristine ng-valid"
                    formcontrolname="batteryListNumber"
                    id="batteryListNumber"
                    name="batteryListNumber"
                    ng-reflect-name="batteryListNumber"
                    type="text"
                  />
                </div>
              </div>
            </div>
          </div>
        </fieldset>
      </div>
    </div>
  </div>
</vtm-battery-list-applicable-edit>
`;
